src/tools/check_versions/src/model.rs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use serde::{Serialize, Deserialize};
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Versions {
+    pub description: String,
+    pub format: String,
+    pub assets: Assets,
+    pub externals: Externals,
+    pub languages: Languages,
+    pub specs: Specs,
+    pub plugins: Plugins
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Assets {
+    pub hypervisor: Hypervisor,
+    pub image: ArchitectureProject,
+    pub initrd: ArchitectureProject,
+    pub kernel: Project,
+    #[serde(rename(deserialize = "kernel-experimental"))]
+    pub kernel_experimental: Project,
+    #[serde(rename(deserialize = "kernel-arm-experimental"))]
+    pub kernel_arm_experimental: Project,
+    #[serde(rename(deserialize = "kernel-dragonball-experimental"))]
+    pub kernel_dragonball_experimental: Project,
+    #[serde(rename(deserialize = "kernel-tdx-experimental"))]
+    pub kernel_tdx_experimental: Project
```

Comment:
I think it would be better if we could find a way to deserialize a generic entry in the versions database. That way we can avoid hard-coding a list of the entries in the code itself. What worries me is that adding a new entry to the database (simple) will require an update to this tool (not simple as you need to know rust).

We have an open PR that does something similar to this:

- https://github.com/kata-containers/kata-containers/pull/6512/files#diff-805954e2eba2cfc43b7cdbbb2f0a286f718287d5a5a8f1ab6f4dd5e2ccc20cadR42-R53

That code is attempting to convert a log entry read from a log file (text) into an `AnyLogMessage` (trait) which is concretely defined (generically) as a `LogMessage`.

/cc @gabevenberg for possible further input.

Reply:
given the presence of the spec in versions.yaml, this should be fairly straightforward.